SocialCareCaseViewerApi.Tests/V1/Gateways/DatabaseGateway.Tests.cs

```diff
@@ -210,8 +208,120 @@ public void CreateWorkerWithInvalidTeamIdThrowsPostWorkerException()
+            act.Should().Throw<GetTeamException>()
+                .WithMessage(
+                    $"Team with Name {createWorkerRequest.Teams[0].Name} and ID {createWorkerRequest.Teams[0].Id} not found");
+        }
+
+        [Test]
+        public void UpdateWorkerThrowsWorkerNotFoundExceptionWhenWorkerDoesNotExist()
+        {
+            var worker = TestHelpers.CreateWorker();
+
+            DatabaseContext.Workers.Add(worker);
+            DatabaseContext.SaveChanges();
+
+            var request = _fixture.Build<UpdateWorkerRequest>()
+                .With(x => x.WorkerId, worker.Id + 1)
+                .Create();
+
+            Action act = () => _classUnderTest.UpdateWorker(request);
+
+            act.Should().Throw<WorkerNotFoundException>()
+                .WithMessage($"Worker with Id {request.WorkerId} not found");
+        }
+
+        [Test]
+        public void UpdateWorkerThrowsPatchWorkerExceptionWhenEmailAddressAlreadyTaken()
+        {
+            var worker1 = TestHelpers.CreateWorker();
+            var worker2 = TestHelpers.CreateWorker();
+
+            DatabaseContext.Workers.Add(worker1);
+            DatabaseContext.Workers.Add(worker2);
+            DatabaseContext.SaveChanges();
+
+            var request = _fixture.Build<UpdateWorkerRequest>()
+                .With(x => x.WorkerId, worker2.Id)
+                .With(x => x.EmailAddress, worker1.Email)
+                .Create();
+
+            Action act = () => _classUnderTest.UpdateWorker(request);
+
+            act.Should().Throw<PatchWorkerException>()
+                .WithMessage($"A worker is already assigned to the email address {request.EmailAddress}");
+        }
+
+        [Test]
+        public void UpdateWorkerUpdatesAnExistingWorkerInDatabase()
+        {
+            var worker = TestHelpers.CreateWorker();
+            var workerTeam = TestHelpers.CreateWorkerTeam(workerId: worker.Id);
+            var team = TestHelpers.CreateTeam(teamId: workerTeam.TeamId);
+            worker.WorkerTeams = new List<WorkerTeam> { workerTeam };
+
+            DatabaseContext.Workers.Add(worker);
+            DatabaseContext.WorkerTeams.Add(workerTeam);
+            DatabaseContext.Teams.Add(team);
+            DatabaseContext.SaveChanges();
+
+            var originalWorker = DatabaseContext.Workers.First(w => w.Id == worker.Id).ShallowCopy();
+            originalWorker.Should().BeEquivalentTo(worker);
+
+            var request = TestHelpers.CreateUpdateWorkersRequest(teamId: team.Id, workerId: worker.Id);
+            _classUnderTest.UpdateWorker(request);
+
+            var updatedWorker = DatabaseContext.Workers.First(w => w.Id == worker.Id);
+
+            updatedWorker.Should().NotBeEquivalentTo(originalWorker);
+
+            CompareUpdatedWorkerAndRequest(updatedWorker, request, worker, team);
+        }
+
+        [Test]
+        public void UpdateWorkerIsActiveFlipsDateStartAndDateEndFromTimeToNull()
```

Comment:
Not close to how workers work but just a thought: is it possible to make a worker go from `isActive: false -> isActive: true`?

Reply:
I added a test for worker going from `false to true` and `true to false` for `isActive`